src/mapping/FGreedyCutSolver.hpp

```diff
@@ -0,0 +1,243 @@
+  for (size_t j = 0; j < greedyIDs.size(); j++)
+  {
+    const auto &y = _inputVertices.at(greedyIDs.at(j)).rawCoords();
+    kernelVector(j) = _basisFunction.evaluate(std::sqrt(computeSquaredDifference(x.rawCoords(), y, _activeAxis))); 
+  }
+}
+
+
+// ---- Initialization ---- //
+
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+template <typename IndexContainer>
+FGreedySolver<RADIAL_BASIS_FUNCTION_T>::FGreedySolver(RADIAL_BASIS_FUNCTION_T basisFunction, const mesh::Mesh &inputMesh, const IndexContainer &inputIDs,
+                                                      const mesh::Mesh &outputMesh, const IndexContainer &outputIDs, std::vector<bool> deadAxis, Polynomial polynomial)
+  : _inSize(inputMesh.vertices().size()), 
+    _outSize(outputMesh.vertices().size()), 
+    _inputVertices(inputMesh.vertices()),
+    _outputVertices(outputMesh.vertices()),
+    _basisFunction(basisFunction)
+{
+  precice::profiling::Event initEvent("FGreedy.initialize", profiling::Synchronize);
+
+  PRECICE_ASSERT(polynomial == Polynomial::OFF, "Poly off");
+  PRECICE_ASSERT(RADIAL_BASIS_FUNCTION_T::isStrictlyPositiveDefinite());
+  PRECICE_ASSERT(_kernelEval.size() == 0); 
+
+  _basisSize = std::min(static_cast<int>(_inSize), _maxIter);
+
+  _activeAxis = std::array<bool, 3>({{false, false, false}});
+  std::transform(deadAxis.begin(), deadAxis.end(), _activeAxis.begin(), [](const auto ax) { return !ax; });
+
+  initEvent.stop();
+}
+
+
+// ---- Evaluation ---- //
+
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+Eigen::VectorXd FGreedySolver<RADIAL_BASIS_FUNCTION_T>::solveConservative(const Eigen::VectorXd &inputData, Polynomial polynomial) const
+{
+  // Not implemented
+  PRECICE_ASSERT(false);
+  return Eigen::VectorXd();
+}
+
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+Eigen::VectorXd FGreedySolver<RADIAL_BASIS_FUNCTION_T>::solveConsistent(Eigen::VectorXd &inputData, Polynomial polynomial) const
+{
+  precice::profiling::Event solveEvent("FGreedy.solveConsistent", profiling::Synchronize);
+
+  Eigen::VectorXd residual = inputData;
+  Eigen::MatrixXd kernelMatrix = Eigen::MatrixXd::Zero(_inSize, _basisSize);
+  Eigen::MatrixXd cut = Eigen::MatrixXd::Zero(_basisSize, _basisSize);
+  std::vector<int> greedyIDs;
+  greedyIDs.reserve(_basisSize);
+  Eigen::VectorXd v = Eigen::VectorXd::Ones(_basisSize);
+  Eigen::VectorXd v2 = Eigen::VectorXd::Ones(_basisSize);
+  Eigen::VectorXd c = Eigen::VectorXd::Zero(_inSize);
+
+  // Iterative selection of new points
+  for (int n = 0; n < _basisSize; ++n) {
+
+    auto [i, fMax] = select(residual);
+    auto x = _inputVertices.at(i); 
+    greedyIDs.push_back(i);
+
+    if (fMax < _tolF) break;
+
+    updateKernelVector(greedyIDs, v, x);
+    v2.head(n) = cut.block(0, 0, n, n).triangularView<Eigen::Lower>() * v.head(n);
+    double p = std::sqrt(v2(n));  
+    v.head(n) = v2.head(n) / p;
+
+    cut.block(n, 0, 1, n).noalias() = -v.block(0, 0, n, 1).transpose() * cut.block(0, 0, n, n).triangularView<Eigen::Lower>();
```

Comment:
```suggestion
    cut.block(n, 0, 1, n).noalias() = -v.head(n).transpose() * cut.block(0, 0, n, n).triangularView<Eigen::Lower>();
```
No? you use this already above? The first cut is essentially one column, cropped to the specific limits of n, right?

Reply:
Using .head(n) produces "error: static assertion failed: YOU_TRIED_CALLING_A_VECTOR_METHOD_ON_A_MATRIX". I assume because, even though it is transposed, the vector is on the left.

Reply:
cut is always a square matrix and gets updated the following way:
![grafik](https://github.com/user-attachments/assets/9086583a-aaab-4344-81d9-735f4bac6bd7)

Reply:
Not sure if this makes any difference, just note that (in such a case) you can also give indices at compile time

https://libeigen.gitlab.io/docs/group__Core__Module.html#gac01f234bce100e39e6928fdc470e5194